walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -62,24 +82,42 @@
     Collections.sort(allBusyTimeRanges, TimeRange.ORDER_BY_START);
 
     busyTimeRanges = getIntersection(busyTimeRanges);
-    allBusyTimeRanges = getIntersection(allBusyTimeRanges);
+    ArrayList<TimeRange> allBusyTimeRangesIntersection = getIntersection(allBusyTimeRanges);
 
     ArrayList<TimeRange> result;
     // if the intersection of all events for both mandatory and optional attendees take up the
     // whole day then just get the free times for the mandatory people
-    if (allBusyTimeRanges.size() == 1 && allBusyTimeRanges.get(0).equals(TimeRange.WHOLE_DAY)) {
+    if (allBusyTimeRangesIntersection.size() == 1 && allBusyTimeRangesIntersection.get(0).equals(TimeRange.WHOLE_DAY)) {
       result = getFreeTime(busyTimeRanges, duration);
     } else {
-      result = getFreeTime(allBusyTimeRanges, duration);
-      if (result.size() == 0) {
-        result = getFreeTime(busyTimeRanges, duration);
+      result = getFreeTime(allBusyTimeRangesIntersection, duration);
+      if (result.size() == 0) { 
+        // solution that optimizes to allow the maximum number of optional attendees to join:
+        ArrayList<TimeRange> cp = new ArrayList<>(allBusyTimeRanges);
+
+        // go through each person and remove all their events until there is enough space
+        for (String entry : optionalAttendeeEvents.keySet()) {
```

Comment:
Does the order in which you go through the attendees matter here?

Imagine this case where all attendees are optional:
The request is for 10 hours, A has a meeting for 6 hours, and B has a meeting for 6 hours, and C has a meeting for 12 hours.
The best thing to do is to schedule in the second half of the day, and leave C out, but if you iterate though attendees in the order A, B, C, I think you'll schedule the meeting in the first half of the day, leaving A & B out.
I might be missing something,  but it would be nice to see a test case like this to prove me wrong. :)
```

    // Events: |-----A-----||------B-----||-------------C------------|
    // Day   : |-----------------------------------------------------|
```

Reply:
Yeah this algorithm is definitely very crude. There's a second part to this algorithm that I did not include in this PR that would theoretically make it better but is currently not behaving as I intended.